agent/host_agent_test.go

```diff
@@ -815,3 +816,64 @@ kovW9X7Ook/tTW0HyX6D6HRciA==
 		})
 	})
 })
+
+var _ = Describe("Agent Unit Tests", func() {
+	Context("When the handleBootstrap func is called", func() {
+		var (
+			bootstrapKubeConf *os.File
+			err               error
+		)
+		BeforeEach(func() {
+			bootstrapKubeConf, err = ioutil.TempFile("", "bootstrap-kubeconfig")
+			Expect(err).NotTo(HaveOccurred())
+			bootstrapKubeConfig = bootstrapKubeConf.Name()
+		})
+		AfterEach(func() {
+			Expect(os.Remove(bootstrapKubeConf.Name())).ShouldNot(HaveOccurred())
+		})
+		It("should return if bootstrap kubeconfig is not valid", func() {
+			testbootstrapKubeconfigInvalid := []byte(`
```

Comment:
Nit: For an invalid kubeconfig, can we just have `testbootstrap = "abc"`?

Reply:
I guess we could use the above as well, I thought of keeping the structure aligned.

Reply:
done